bravado/mapping/unmarshal.py

```diff
@@ -0,0 +1,143 @@
+import jsonschema
+
+from bravado.exception import SwaggerError
+from bravado.mapping import formatter, schema
+from bravado.mapping.model import is_model, MODEL_MARKER
+from bravado.swagger_type import SWAGGER20_PRIMITIVES, is_list_like, \
+    is_dict_like
+
+
+def unmarshal_schema_object(swagger_spec, schema_object_spec, value):
+    """
+    Unmarshal the value using the given schema object specification.
+
+    Unmarshaling includes:
+    - validate that the value conforms to the schema_object_spec
+    - transform the value according to 'format' if available
+    - return the value in a form suitable for use. e.g. conversion to a Model
+      type.
+
+    :type swagger_spec: :class:`bravado.mapping.spec.Spec`
+    :type schema_object_spec: dict
+    :type value: int, float, long, string, unicode, boolean, list, dict
+    :return: unmarshaled value
+    :rtype: int, float, long, string, unicode, boolean, list, dict, Model type
+    """
+    obj_type = schema_object_spec['type']
+
+    if obj_type in SWAGGER20_PRIMITIVES:
+        return unmarshal_primitive(schema_object_spec, value)
+
+    elif obj_type == 'array':
+        return unmarshal_array(swagger_spec, schema_object_spec, value)
+
+    elif is_model(schema_object_spec):
```

Comment:
Is the difference between a model and an object is that model is defined as a `ref` in #/definitions whereas object is defined inline?